lib/core/request.js

```diff
@@ -74,6 +75,8 @@ class Request {
         // TODO: Ignore error if body has ended?
         this.onError(err)
       })
+    } else if (util.isGenerator(body)) {
+      this.body = Readable.from(body())
     } else if (util.isBuffer(body)) {
```

Comment:
We could also add support for iterable at the same time?

Reply:
Ahh so checking for `[Symbol.iterator]` or something like that right?

Reply:
check what Readable.from does

Reply:
yep so in that case if its an iterable object then we can pass it directly to Readable.from

if its a generator function we have to pass the return value of it to Readable.from